pkg/discovery/service.go

```diff
@@ -144,8 +160,39 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 	return nil
 }
 
+// graveyard attempts to ping all nodes in the Seen() map from Kademlia and adds them to the cache
+// if they respond. This is an attempt to resurrect nodes that may have gone offline in the last hour
+// and were removed from the cache due to an unsuccessful response.
+func (discovery *Discovery) graveyard(ctx context.Context) error {
+	discovery.log.Debug("starting node graveyard resurrection")
+	seen := discovery.kad.Seen()
```

Comment:
how is the graveyard discovery.kad.Seen() different from refresh's discovery.kad.Seen()? Are they looking at the same overlay cache?  graveyard()'s overlay cache(which holds non-pingable nodes) shouldn't it be different overlay cache from the refresh overlay cache?